### DISCUSSION TITLE
How to trigger a re-positioning of a Dialog?

### DISCUSSION THREAD

**Discussion by cedeber** [adobe/react-spectrum#3471] [Q&A] [answered]
I am currently loading the content of a Dialog/Modal asynchronously. It is build thanks to `useOverlayTriggerState`, `useOverlayTrigger`, `useOverlay `and `useModal`. Is it first placed correctly with the Spinner, but once the content is added to the Dialog, the left position is kept and then the Dialog is misplaced. The dialog is rendered inside of a `OverlayContainer`.

I control the state by doing some `state.open` and `close` by code instead of using the `triggerProps` pointer events. The `triggerRef ` is still given for the positioning and it works as expected, at least with the spinner.

Hard for me to create an example here, but:
- will a dialog be replaced automatically if the content change. I mean managed by react-aria, somehow? I know the height doesn't work neither because we have some modal with tabs and different content heights.
- and if not, does it exist a trick to trigger a position calculation for a modal by code?

Thanks.

**Comment by snowystinger** [comment, marked as answer]
Looks like we expose a function on the return of useOverlayPosition `updatePosition` https://react-spectrum.adobe.com/react-aria/useOverlayPosition.html
Can that do what you need?

**Reply by cedeber** [reply]
Looks good so far, yes, thanks.
